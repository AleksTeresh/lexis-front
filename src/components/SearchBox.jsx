/* @flow */

import React from 'react'
import styled from 'styled-components'

import Paper from 'material-ui/Paper'
import TextField from 'material-ui/TextField'

import { Icon, CenterBox } from '.'

const Container = styled(Paper)`
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: white;
  border-radius: 4px;
  padding: 0 2rem 0 1rem;
  width: 100%;
`

type SearchBoxProps = {
  text: string,
  onChange: (val: string) => void
}

const SearchBox = ({ text, onChange }: SearchBoxProps) => (
  <Container elevation={3}>
    <Icon iconClassName='fa fa-search' size='1.5em' spacing={'18px'} />
    <TextField
      id='full-width'
      placeholder='Search'
      value={text}
      onChange={ev => onChange(ev.target.value)}
      fullWidth
      margin='normal'
    />
  </Container>
)

export default SearchBox
