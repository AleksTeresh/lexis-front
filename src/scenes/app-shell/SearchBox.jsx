import React from 'react'
import styled from 'styled-components'
import { Text, Icon, CenterBox } from 'common-components'

const Container = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: white;
  border-radius: 4px;
  padding: 0 1rem 0 1rem;
  width: 100%;
`
const SearchBox = props => (
  <Container>
    <Icon iconClassName='fa fa-search' size='1.5em' spacing={'18px'} />
    <Text primary normal fontSize={'1.2rem'}>
      {' '}
      Type to search{' '}
    </Text>
  </Container>
)

export default SearchBox
