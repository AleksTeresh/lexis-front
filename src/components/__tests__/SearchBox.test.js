/* eslint-env jest */

import React from 'react'
import SearchBox from '../SearchBox'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'

configure({ adapter: new Adapter() })

describe('SearchBox Component', () => {
  it('renders a snapshot', () => {
    const wrapper = shallow(<SearchBox />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
