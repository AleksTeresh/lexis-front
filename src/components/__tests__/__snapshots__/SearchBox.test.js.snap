// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Component renders a snapshot 1`] = `
<Styled(withStyles(Paper))
  elevation={3}
>
  <Icon
    iconClassName="fa fa-search"
    size="1.5em"
    spacing="18px"
  />
  <TextField
    fullWidth={true}
    id="full-width"
    margin="normal"
    onChange={[Function]}
    placeholder="Search"
    required={false}
    select={false}
  />
</Styled(withStyles(Paper))>
`;
